### PR TITLE
feat(components/datetime): add datepicker test harness

### PR DESCRIPTION
**Pull request by Blackbaud-SandhyaRajasabeson**
[AB#2195418](https://dev.azure.com/blackbaud/Products/_boards/board/t/SKY%20UX%20Program/Stories/?workitem=2195418)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2836/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2836/storybooks/ag-grid-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2836/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2836/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2836/playground/)

File: apps/code-examples/src/app/code-examples/datetime/datepicker/basic/demo.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { SkyDatepickerHarness } from '@skyux/datetime/testing';
+import { SkyInputBoxHarness } from '@skyux/forms/testing';
+
+import { DemoComponent } from './demo.component';
+
+fdescribe('Basic datepicker demo', () => {
```
Comment (Blackbaud-CoreyArcher) [inline apps/code-examples/src/app/code-examples/datetime/datepicker/basic/demo.component.spec.ts]: `fdescribe`

File: libs/components/datetime/testing/src/datepicker/datepicker-harness.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+   */
+  public async isDatepickerOpen(): Promise<boolean> {
+    return (
+      (await (
+        await this.#getCalendarButton()
+      ).getAttribute('aria-expanded')) === 'true'
+    );
+  }
+
+  private async getAriaControls(): Promise<string | null> {
```
Comment (Blackbaud-CoreyArcher) [inline libs/components/datetime/testing/src/datepicker/datepicker-harness.ts]: Is there a reason this is `private`?

File: libs/components/datetime/testing/src/datepicker/datepicker-calendar-harness.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+  #getDaypicker = this.locatorForOptional('sky-daypicker');
+  #getMonthpicker = this.locatorForOptional('sky-monthpicker');
+  #getNextButton = this.locatorFor('.sky-datepicker-btn-next');
+  #getPreviousButton = this.locatorFor('.sky-datepicker-btn-previous');
+  #getSelected = this.locatorFor('.sky-datepicker-btn-selected');
+  #getTitle = this.locatorFor('.sky-datepicker-calendar-title > strong');
+  #getTitleButton = this.locatorFor('.sky-datepicker-calendar-title');
+
+  public static with(
```
Comment (Blackbaud-CoreyArcher) [inline libs/components/datetime/testing/src/datepicker/datepicker-calendar-harness.ts:26]: Docs comment?

File: libs/components/datetime/testing/src/datepicker/datepicker-harness.ts (hunk 2 above)
Comment (Blackbaud-SandhyaRajasabeson) [reply libs/components/datetime/testing/src/datepicker/datepicker-harness.ts]: yeah this is just needed to get the calendar but consumers dont need it

File: apps/code-examples/src/app/code-examples/datetime/datepicker/basic/demo.component.spec.ts (hunk 1 above)
Comment (Blackbaud-SandhyaRajasabeson) [reply apps/code-examples/src/app/code-examples/datetime/datepicker/basic/demo.component.spec.ts]: im surprised this didnt get caught with the build 🤔

File: libs/components/datetime/testing/src/datepicker/datepicker-calendar-harness.ts (hunk 3 above)
Comment (Blackbaud-SandhyaRajasabeson) [reply libs/components/datetime/testing/src/datepicker/datepicker-calendar-harness.ts:26]: honestly i think this should be internal bc consumers would nevber be able to use this

**Review by Blackbaud-CoreyArcher — DISMISSED** [review]

**Review by Blackbaud-CoreyArcher — APPROVED** [review]

**Pull request merged**